File: app/src/test/java/com/worldventures/dreamtrips/BaseTest.java
package com.worldventures.dreamtrips;

import com.worldventures.core.janet.cache.CacheResultWrapper;
import com.worldventures.core.test.common.RxJavaSchedulerInitializer;

import org.junit.Before;
import org.junit.platform.runner.JUnitPlatform;
import org.junit.runner.RunWith;
import org.mockito.MockitoAnnotations;

import io.techery.janet.ActionService;

@RunWith(JUnitPlatform.class)
public abstract class BaseTest {

   static {
      RxJavaSchedulerInitializer.init();
   }

   @Before
   public void setupBase() {
      MockitoAnnotations.initMocks(this);
   }

   protected static CacheResultWrapper cachedService(ActionService service) {
      return new CacheResultWrapper(service);
   }
}

File: wallet/src/test/java/com/worldventures/wallet/ui/common/BaseTest.kt
package com.worldventures.wallet.ui.common

import com.trello.rxlifecycle.RxLifecycle
import com.worldventures.core.test.common.AndroidRxJavaSchedulerInitializer
import com.worldventures.core.test.common.RxJavaSchedulerInitializer
import com.worldventures.wallet.ui.common.base.screen.RxLifecycleView
import org.mockito.Mockito
import org.mockito.MockitoAnnotations
import rx.subjects.BehaviorSubject

abstract class BaseTest {

   init {
      MockitoAnnotations.initMocks(this)
   }

   companion object {
      init {
         AndroidRxJavaSchedulerInitializer.init()
         RxJavaSchedulerInitializer.init()
      }

      fun <V : RxLifecycleView> mockScreen(clazz: Class<V>) : V {
         val screen : V = Mockito.mock(clazz)
         Mockito.`when`(screen.bindUntilDetach<Any>()).thenReturn(RxLifecycle.bind(BehaviorSubject.create<Any>()))
         return screen
      }
   }
}
